make-rules/setup.py.mk

```diff
@@ -64,46 +64,31 @@ COMPONENT_TEST_ENV += $(PYTHON_ENV)
 # Reset arguments specified as environmnent variables
 COMPONENT_BUILD_ARGS =
 
-# Make sure the default python version is built and installed last and so is
-# the canonical version.
-define python-order-rule
-$(BUILD_DIR)/%-$(PYTHON_VERSION)/.built:     $(BUILD_DIR)/%-$(1)/.built
-$(BUILD_DIR)/%-$(PYTHON_VERSION)/.installed: $(BUILD_DIR)/%-$(1)/.installed
-endef
-
-$(foreach pyver,$(filter-out $(PYTHON_VERSION),$(PYTHON_VERSIONS)),$(eval $(call python-order-rule,$(pyver))))
-
-# Create a distutils config file specific to the combination of build
-# characteristics (bittedness x Python version), and put it in its own
-# directory.  We can set $HOME to point distutils at it later, allowing
-# the install phase to find the temporary build directories.
-CFG=.pydistutils.cfg
-$(BUILD_DIR)/config-%/$(CFG):
-	$(MKDIR) $(@D)
-	echo "[build]\nbuild_base = $(BUILD_DIR)/$*" > $@
+# We need to copy the source dir to avoid its modification by install target
+# where egg-info is re-generated
+CLONEY_ARGS = CLONEY_MODE="copy"
 
 # build the configured source
-$(BUILD_DIR)/%/.built:	$(SOURCE_DIR)/.prep $(BUILD_DIR)/config-%/$(CFG)
+$(BUILD_DIR)/%/.built:	$(SOURCE_DIR)/.prep
 	$(RM) -r $(@D) ; $(MKDIR) $(@D)
+	$(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
 	$(COMPONENT_PRE_BUILD_ACTION)
```

Comment:
Is $(ENV) $(CLONEY_ARGS) $(CLONEY) the correct order?

Reply:
```
$ grep CLONEY make-rules/*
make-rules/ant.mk:      $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/attpackagemake.mk:   $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/font.mk:     $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/justmake.mk: $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/makemaker.mk:        $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/modulebuild.mk:      $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/shared-macros.mk:CLONEY =    $(WS_TOOLS)/cloney
make-rules/waf.mk:      $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/waf.mk:# means $(CLONEY) has already run.  System-test needs cloning but not
make-rules/waf.mk:# but since we don't have that, we depend on .prep and run $(CLONEY) here.
make-rules/waf.mk:      $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
make-rules/waf.mk:      $(ENV) $(CLONEY_ARGS) $(CLONEY) $(SOURCE_DIR) $(@D)
$
```

Reply:
Ok, I was confused :-)